### PR TITLE
[Bugfix] Fix bugs and clean code for LANS

### PR DESCRIPTION
**Pull request by zhen-jia**
## Description ##
Add a group_cast and clean code for LANS, which only takes FP32.

## Checklist ##

- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

cc @awslabs/raf-reviewer

### CODE REVIEW

File: src/op/dialect/cuda/lans.cc [hunk 1]
```diff
@@ -38,7 +37,7 @@ class LansImpl : public raf::op::OpEnv {
     DLTensor* t0 = ir::Downcast<TensorValue>(args->tensor_list[0]);
     auto datatype = t0->dtype;
     CHECK(datatype.code == kDLFloat);
-    CHECK((datatype.bits == 32) || (datatype.bits == 16));
+    CHECK((datatype.bits == 32)) << "LANS only takes FP32 inputs";
```
Comment (comaniac) [inline src/op/dialect/cuda/lans.cc:40]: Accuracy issue?

**Review by comaniac — COMMENTED** [review]
@whbldhwj please also help review this PR.

File: src/op/dialect/cuda/lans.cc (hunk 1 above)
Comment (comaniac) [reply src/op/dialect/cuda/lans.cc:40]: Got it.

**Review by whbldhwj — APPROVED** [review]
LGTM

**Comment by comaniac** [comment]
Thanks @zhen-jia @whbldhwj

**Pull request merged**